tests/test_dcos_e2e/docker_utils/test_loopback.py

```diff
@@ -18,21 +18,34 @@ def test_loopback(self) -> None:
         client = docker.from_env(version='auto')
 
         with DockerLoopbackVolume(size=1) as device:
-            block_device_exists_cmd = ['lsblk', device.path]
             new_container = client.containers.create(
                 privileged=True,
                 detach=True,
                 image='centos:7',
             )
             new_container.start()
+
+            block_device_exists_cmd = ['lsblk', device.path]
             exit_code, output = new_container.exec_run(
                 cmd=block_device_exists_cmd,
             )
-            new_container.stop()
-            new_container.remove()
+            assert exit_code == 0, device.path + ': ' + output.decode()
 
+            block_device_has_right_size = [
+                '[',
+                '$(stat -f "%z" ' + device.path + ')',
+                '-eq',
+                '"1048576"',
+                ']',
+            ]
+            exit_code, output = new_container.exec_run(
+                cmd=block_device_has_right_size,
+            )
             assert exit_code == 0, device.path + ': ' + output.decode()
 
+            new_container.stop()
```

Comment:
Let's move these back above the assertion. That means that even if the assertion fails, the container will be removed.

Reply:
Good point! Though I'd rather use a `try... finally` block here, as there's more than one assertion that could fail.

Reply:
Either works for me.